vllm/model_executor/layers/fused_moe/rocm_aiter_fused_moe.py

```diff
@@ -39,6 +40,16 @@ def rocm_aiter_fused_experts(
     from vllm.model_executor.layers.quantization.utils.fp8_utils import (
         per_token_group_quant_fp8)
 
+    if apply_router_weight_on_input:
+        _, topk = topk_weights.shape
+        assert (
+            topk == 1
+        ), "Only support topk=1 when `apply_router_weight_on_input` is True"
+
+        hidden_states = hidden_states * topk_weights.to(hidden_states.dtype)
+        topk_ids = topk_ids.to(torch.int32)
+        topk_weights = torch.ones_like(topk_weights, dtype=torch.float32)
```

Comment:
Does AITER require fp32 weight?

Reply:
yes, the passed-in `topk_weights` must in fp32 dtype, otherwise, it will have numeric issues